modular_skyrat/modules/events/code/modules/events/apc_failure.dm

```diff
@@ -0,0 +1,22 @@
+/datum/round_event_control/apc_failure
+	name = "APC Failure"
+	typepath = /datum/round_event/apc_failure
+	weight = 50
+	max_occurrences = 5 // want this to be rare, so as to not piss off engies
+	alert_observers = FALSE
+
+/datum/round_event/apc_failure
+	fakeable = FALSE
+
+/datum/round_event/apc_failure/start()
+	var/obj/machinery/power/apc/target_apc = pick(GLOB.apcs_list)
+	if(is_station_level(target_apc.z))
```

Comment:
And if its not on the station z level it does nothing?

Events that do nothing aren't good.

Reply:
I'll come back to this tomorrow cause I'm far too tired to think of a solution